quic/s2n-quic-transport/src/space/datagram.rs

```diff
@@ -109,3 +116,111 @@ impl<'a, C: WriteContext> s2n_quic_core::datagram::Packet for Packet<'a, C> {
         self.datagrams_prioritized
     }
 }
+
+#[test]
+// Check that our default on_transmit function doesn't continue to pop datagrams
+// off the send queue if the remaining packet space is too small to send datagrams.
+fn has_written_test() {
```

Comment:
This seems to be testing the default sender, rather than the datagram manager. I think this belongs there, right?

Reply:
Well, yes and no. Technically, it's testing the default_sender's implementation of on_transmit and on_transmit needs a WriteContext in order to work, which lives in s2n-quic-transport. And since default.rs is in s2n-quic-core, it can't depend on anything in s2n-quic-transport.

Reply:
Doesn't it only need an implementation of `Packet`?

Reply:
Right, which has a WriteContext in it.

Reply:
No `Packet` is a trait which can be implemented by anything. So you can write your own mock `Packet` struct